### ISSUE TITLE
Does not compile

### ISSUE THREAD

**Issue by nils-eilers** [mburkley/tms9900-gcc#69]
Tried compilation under Arch Linux but it failed. Any alternatives or suggestions?

`make[2]: Entering directory '/home/j/source-codes/tms9900-gcc/build/binutils-2.19.1/libiberty'

...

.././libiberty/regex.c: In function 'byte_regex_compile':
.././libiberty/regex.c:2278:11: error: implicit declaration of function 'free' [-Wimplicit-function-declaration]
...
.././libiberty/regex.c:4835:11: error: implicit declaration of function 'abort' [-Wimplicit-function-declaration]

...

make[2]: *** [Makefile:925: regex.o] Error 1
make[2]: Leaving directory '/home/j/source-codes/tms9900-gcc/build/binutils-2.19.1/libiberty'
make[1]: *** [Makefile:8670: all-libiberty] Error 2
make[1]: Leaving directory '/home/j/source-codes/tms9900-gcc/build/binutils-2.19.1'
make: *** [Makefile:717: all] Error 2
=== Failed to build Binutils ===`

**Comment by mburkley** [comment]
I'm afraid I haven't seen that issue.  It looks like some kind of configuration problem.  The GCC patch doesn't make any changes to regex.c so maybe it would be interesting to test building binutils without the patches?  It is possible that binutils-2.19.1 is now so old that it won't reliably compiler and build on arch linux.

**Comment by nils-eilers** [comment]
Thanks. I tried a stock binutils-2.19.1 but that didn't compile either. I then moved on from Arch to Windows 10 WSL2 (aka Ubuntu 22.04.5 LTS) and that did the job. I assume, Arch Linux uses newer versions of whatever which introduced an incompability, so we might see this issue sooner or later on Ubuntu as well.

**Comment by mburkley** [comment]
Yes that's what I expected unfortunately.  The long term fix is to upgrade to a later gcc and binutils but I haven't had much time to work on them lately.

**Comment by 2mac** [comment]
I'm running into this issue on Fedora 42 as well. Looking at the source, these bad declarations are in a preprocessor block that will be included only if neither `STDC_HEADERS` nor `_LIBC` are defined. `STDC_HEADERS` is set by the autoconf check for ANSI C header files. To my knowledge, these should all be included with gcc, but for some reason autoconf isn't finding them.

The bad definitions in regex.c are arguably bugs of binutils, but maybe it is worth investigating the autoconf problem and patching it if it's not fixed in a newer binutils version.
